rules-reviewed/eap7/eap71/hibernate51-53.windup.xml

```diff
@@ -119,17 +119,23 @@
             </perform>
         </rule>
         -->
-        <!--https://issues.jboss.org/browse/WINDUPRULE-387
+        <!--https://issues.jboss.org/browse/WINDUPRULE-387-->
         <rule id="hibernate51-53-01100">
             <when>
+                <and>
+                    <javaclass references="javax.persistence.TableGenerator">
+                        <location>IMPORT</location>
+                    </javaclass>
+                </and>
             </when>
             <perform>
+                <hint title="Hibernate 5.3  - @TableGenerator changes interpretation" effort="1" category-id="potential">
+                    <message>Existing applications migrating to 5.3 and using the `@TableGenerator` have to set the `hibernate.id.generator.stored_last_used` configuration property to `false`.</message>
+                    <link title="Red Hat JBoss EAP 7.2: Migrating from Hibernate ORM 5.1 to Hibernate ORM 5.3" href="https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.2/html-single/migration_guide/#migrating_from_hibernate_5_1_to_5_3" />
```

Comment:
There's a link to the list of issues where the user can find this issue more easily
```suggestion
                    <link title="Red Hat JBoss EAP 7.2: Migrating from Hibernate ORM 5.1 to Hibernate ORM 5.3" href="https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.2/html-single/migration_guide/#hibernate_orm_5_3_features" />
```

Reply:
I applied what's in the Phil's rule.
I can change it though.

Reply:
That's fine the link you used because it's the one to the begin of the hibernate section.
In this case it seems to me that the link can be more precise and bring the user closer to the documentation so it can be updated.
This is also something to check for other rules anyway since we started from the same "template"